src/components/Inputs/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,269 @@
+import React, { FC, useState } from 'react';
+import { ButtonSize, DefaultButton } from '../../Button';
+import { Icon, IconName, IconSize } from '../../Icon/index';
+import {
+    InputWidth,
+    TextInputProps,
+    TextInputShape,
+    TextInputTheme,
+} from '../index';
+import { Tooltip } from '../../Tooltip';
+import { useDebounce } from '../../../hooks/useDebounce';
+import { classNames, uniqueId } from '../../../shared/utilities';
+
+import styles from '../input.module.scss';
+
+export const TextInput: FC<TextInputProps> = ({
+    allowDisabledFocus = false,
+    ariaLabel,
+    autoFocus = false,
+    className,
+    clearButtonAriaLabel,
+    disabled = false,
+    htmlType = 'text',
+    iconProps,
+    iconButtonProps,
+    id,
+    inputWidth = InputWidth.fitContent,
+    label,
+    labelIconButtonProps,
+    maxlength,
+    minlength,
+    name,
+    numbersOnly = false,
+    onBlur,
+    onChange,
+    onFocus,
+    onKeyDown,
+    placeholder,
+    required = false,
+    shape = TextInputShape.Rectangle,
+    style,
+    theme = TextInputTheme.light,
+    value,
+    waitInterval = 10,
+}) => {
+    const [clearButtonShown, setClearButtonShown] = React.useState(false);
+    const [inputId] = useState(uniqueId('input-'));
+    const inputField: HTMLElement = document.getElementById(id ? id : inputId);
+
+    const iconClassNames: string = classNames([
+        styles.iconWrapper,
+        styles.leftIcon,
+    ]);
+
+    const iconButtonClassNames: string = classNames([
+        styles.iconButton,
+        styles.leftIcon,
+    ]);
+
+    const textInputClassNames: string = classNames([
+        className,
+        {
+            [styles.withIcon]:
+                !!iconProps &&
+                !!iconProps.path &&
+                shape === TextInputShape.Rectangle,
```

Comment:
```suggestion
                !!iconProps?.path &&
                shape === TextInputShape.Rectangle,
```
SAme on line 72, 88,